### ISSUE TITLE
Add support for mult-gpu (single node) prompt tuning

### ISSUE THREAD

**Issue by gkumbhat** [caikit/caikit-nlp#32]
## Is your feature request related to a problem? Please describe.

Currently prompt tuning module only support training on single gpu. In order to support larger models, or larger batch sizes or allow larger text sizes, we would need to provide support to handle training across multiple GPUs (on a single node).

## Describe the solution you'd like

Support for multi-gpu training in prompt tuning module.

## Describe alternatives you've considered

- Increasing `max_length` by reducing batch size - however, this can have impact on quality and would require tuning other parameters such as learning rate and accumulation step, which is doable, but would require more fiddling. Other than that, it also still doesn't allow us to prompt-tune larger models which won't fit in 1 GPU.

## Additional context

Add any other context about the feature request here.

**Comment by chakrn** [comment]
Redundant - Closing
